### PR TITLE
DaughterSwap Support | Twistys/FakeHub/SexyHub/PropertySex/Babes…

### LINKED ISSUE #498: Fake Taxi/Fake Hub not returning results
Site: Fake Taxi/Faketaxi.com
File Name: ftx_hazel_jack_s2_RU122314_1080p_6000
Search Term: Fake Taxi 4076331
Logs: attached

This was working as of two or three days ago, now it's not returning anything. In the logs it looks like Fake Taxi is being shortened to ake taxi, but I'm not sure if that's the same as when it was working. I think the relevant bits of the log are only through 126 or so, stuff after that was just me confirming other sites worked and trying a few things. No luck.
[com.plexapp.agents.phoenixadult.log](https://github.com/PAhelper/PhoenixAdult.bundle/files/3974571/com.plexapp.agents.phoenixadult.log)

### LINKED ISSUE #515: 18OnlyGirls list index out of range
```
2019-12-30 02:41:33,018 (153225e44700) :  INFO (__init__:44) - *******MEDIA TITLE****** 18onlygirls 16 05 17 Melena Maria
2019-12-30 02:41:33,018 (153225e44700) :  INFO (__init__:51) - Getting Search Settings for: 18onlygirls 16 05 17 Melena Maria
2019-12-30 02:41:33,018 (153225e44700) :  INFO (PAsearchSites:957) - mediaTitle w/ possible abbrieviation: 18onlygirls 16 05 17 Melena Maria
2019-12-30 02:41:33,028 (153225e44700) :  INFO (PAsearchSites:1146) - mediaTitle w/ possible abbrieviation fixed: 18onlygirls 16 05 17 Melena Maria
2019-12-30 02:41:33,031 (153225e44700) :  INFO (PAsearchSites:1161) - ^^^^^^^ siteID: 794
2019-12-30 02:41:33,032 (153225e44700) :  INFO (PAsearchSites:1163) - ^^^^^^^ Shortening Title
2019-12-30 02:41:33,032 (153225e44700) :  INFO (PAsearchSites:1164) - 18onlygirls vs 18onlygirls
2019-12-30 02:41:33,032 (153225e44700) :  INFO (PAsearchSites:1168) - 1
2019-12-30 02:41:33,032 (153225e44700) :  INFO (PAsearchSites:1187) - 7
2019-12-30 02:41:33,032 (153225e44700) :  INFO (PAsearchSites:1223) - searchTitle (before date processing): 16 05 17 Melena Maria
2019-12-30 02:41:33,033 (153225e44700) :  INFO (__init__:61) - Search Title: Melena Maria
2019-12-30 02:41:33,033 (153225e44700) :  INFO (__init__:65) - Search Date: 2016-05-17
2019-12-30 02:41:33,033 (153225e44700) :  INFO (__init__:70) - Melena%20Maria
2019-12-30 02:41:33,033 (153225e44700) :  INFO (siteGirlsWay:9) - SceneID: Melena
2019-12-30 02:41:33,033 (153225e44700) :  INFO (siteGirlsWay:14) - Scene Title: Maria
2019-12-30 02:41:33,035 (153225e44700) :  DEBUG (networking:138) - Fetching 'http://18onlygirls.tv/?s=Melena' from the HTTP cache
2019-12-30 02:41:33,038 (153225e44700) :  CRITICAL (agentkit:1014) - Exception in the search function of agent named 'PhoenixAdult', called with keyword arguments {'year': None, 'id': '130767', 'name': '18onlygirls 16 05 17 Melena Maria'} (most recent call last):
  File "/usr/lib/plexmediaserver/Resources/Plug-ins-ac2afe5f8/Framework.bundle/Contents/Resources/Versions/2/Python/Framework/api/agentkit.py", line 1007, in _search
    agent.search(*f_args, **f_kwargs)
  File "/config/Library/Application Support/Plex Media Server/Plug-ins/PhoenixAdult.bundle/Contents/Code/__init__.py", line 117, in search
    results = PAsearchSites.siteGirlsWay.search(results, encodedTitle, title, searchTitle, siteNum, lang, searchByDateActor, searchDate, searchSiteID)
  File "/config/Library/Application Support/Plex Media Server/Plug-ins/PhoenixAdult.bundle/Contents/Code/siteGirlsWay.py", line 17, in search
    titleNoFormatting = searchResult.xpath('//div[@class="titleBar"]/h1')[0].text_content().strip()
  File "/usr/lib/plexmediaserver/Resources/Plug-ins-ac2afe5f8/Framework.bundle/Contents/Resources/Versions/2/Python/Framework/code/sandbox.py", line 108, in <lambda>
    _getitem_           = lambda x, y: x.__getitem__(y),
IndexError: list index out of range
```

### PR DESCRIPTION
**Pull request by HWatBobby**
DaughterSwap:
- Now Supported.

Twistys/FakeHub/SexyHub/PropertySex/Babes/FamilyHookups:
- Collection/Subsite bugfix. Resolves #498, #509, #516, #513, #504.

18OnlyGirls:
- Bugfix. Resolves #515.

Erito:
- Now Supported.

TrueAmateurs: 
- Now Supported.

### CODE REVIEW

**Comment by HWatBobby** [comment]
I'll also be working to fix Twistys, FakeHub, and any other site broken on PornPortal. Mofos and Babes are already fixed in last merge. Will be testing Twistys and FakeHub soon.

**Comment by HWatBobby** [comment]
Ignore the Create and Delete DaughterSwap commits. Still learning.

**Comment by pedrolara-boop** [comment]
Tested Twistys and it works!

**Comment by HWatBobby** [comment]
I believe this PR should be good unless anyone knows of a related site that could be quickly fixed. I've tested all sites, with the exception if SexyHub since I do not have any scenes from that site, and they all work. Changes made to SexyHub are same as other site fixes.

@claygoldfinch If you want to review let me know if you see any issues

File: Contents/Code/networkSexyHub.py [hunk 1]
```diff
@@ -8,15 +8,15 @@ def search(results,encodedTitle,title,searchTitle,siteNum,lang,searchByDateActor
     Log("SceneID: " + sceneID)
     try:
         sceneTitle = encodedTitle.split('%20', 1)[1].replace('%20',' ')
-    except:
+except:
```
Comment (claygoldfinch) [inline Contents/Code/networkSexyHub.py]: Indent

File: Contents/Code/siteTrueAmateurs [hunk 2]
```diff
@@ -0,0 +1,123 @@
+import PAsearchSites
```
Comment (claygoldfinch) [inline Contents/Code/siteTrueAmateurs]: Add a .py extension to the the siteTrueAmateurs file

**Comment by claygoldfinch** [comment]
Out of town till later tonight -  I'll have this fully tested by Monday. Sorry for the delay!

File: Contents/Code/siteDaughterSwap.py [hunk 3]
```diff
@@ -1,94 +1,96 @@
...
+def search(results,encodedTitle,title,searchTitle,siteNum,lang,searchByDateActor,searchDate,searchSiteID):
+    if searchSiteID != 9999:
+        siteNum = searchSiteID
+    searchString = searchTitle.lower().replace(" ","-")
+    Log("searchString: " + searchString)
+    url = PAsearchSites.getSearchSearchURL(siteNum) + searchString
+    searchResult = HTML.ElementFromURL(url)
+    titleNoFormatting = searchResult.xpath('//p[@class="video-title hidden-xs"]')[0].text_content().strip()
+    curID = url.replace('/','_').replace('?','!')
+    subSite = "DadCrush"
```
Comment (claygoldfinch) [inline Contents/Code/siteDaughterSwap.py]: Update to "DaughterSwap".

**Comment by claygoldfinch** [comment]
Looks great, thanks for all the hard work! Happy to have another active contributor.

**Pull request merged**